crates/vm/src/arch/vm.rs

```diff
@@ -851,7 +835,7 @@ where
     }
 
     /// Executes a program, compute the trace heights, and returns the public values.
-    pub fn execute_with_max_heights_and_compute_heights(
+    pub fn execute_and_compute_heights(
```

Comment:
why is it computing heights but also taking in `max_trace_heights`

Reply:
this is probably an artifact of the past. let me check if just using `max_trace_heights` works

Reply:
ah no. `compute_height` returns this `VmComplexTraceHeights` struct. we can probably replace it with a `Vec` of trace heights, but it'll require more changes